pkgs/by-name/ba/bambu/package.nix

```diff
@@ -0,0 +1,96 @@
+{ stdenv
+, pkgsi686Linux
+, makeWrapper
+, lib
+, glibc
+, bambu-unwrapped
+, verilator
+, bash
+, gawk
+, coreutils
+, findutils
+, gnumake
+, gnused
+, gnugrep
+, verilog
+, which
+}:
+
+stdenv.mkDerivation
+{
+  pname = "bambu";
+  inherit (bambu-unwrapped) meta version;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  doCheck = true;
+
+  unpackPhase = ":";
+
+  buildPhase = ''
+    runHook preBuild
+
+    mkdir $out
+    ln -s ${bambu-unwrapped}/* $out
+    rm $out/bin
+    mkdir $out/bin
+    ln -s ${bambu-unwrapped}/bin/* $out/bin
+    for f in $out/bin/*; do
+      if [ -f "$f" ]; then
+          wrapProgram "$f" \
+            --suffix LDFLAGS " " "-L${glibc.static}/lib" \
+            --set PATH ${lib.makeBinPath [
+              pkgsi686Linux.gcc
+              verilator
+              bash
+              gawk
+              findutils
+              coreutils
+              gnumake
+              gnused
+              gnugrep
+              verilog
+              which
+            ]}
+      fi
+    done
+
+    runHook postBuild
+  '';
+
+  checkPhase = ''
+    runHook preCheck
+
+    # Check that bambu can synthesize and verilate a simple program
+    cat <<EOT >> min_max.cpp
+    void min_max(int* numbers, int numbers_length, int* out_max, int* out_min)
+    {
+       int local_max = numbers[0];
+       int local_min = numbers[0];
+       int i = 0;
+       for(i = 0; i < numbers_length; i++)
+       {
+          if(numbers[i] > local_max)
+          {
+             local_max = numbers[i];
+          }
+          if(numbers[i] < local_min)
+          {
+             local_min = numbers[i];
+          }
+       }
+       *out_max = local_max;
+       *out_min = local_min;
+    }
+    EOT
+    cat <<EOT >> testbench.xml
+    <?xml version="1.0"?>
+    <function>
+      <testbench numbers_length="5" numbers="{31, 73, 69, 54, 10}" out_max="{0}" out_min="{0}" />
+    </function>
+    EOT
+    $out/bin/bambu min_max.cpp --top-fname=min_max --generate-tb=testbench.xml --simulate --simulator=VERILATOR -v4 --compiler=I386_CLANG16 -O5
```

Comment:
Ofborg build fails https://logs.ofborg.org/?key=nixos/nixpkgs.244062&attempt_id=e62a3879-58e7-47a5-bfff-09d84573c94c

```
....
kill failed: No such process
ERROR: MDPI driver: Abrupt exception: 6
ERROR: MDPI driver: Unable to send signal.
kill failed: No such process
ERROR: MDPI driver: Abrupt exception: 6
ERROR: MDPI driver: Unable to send signal.
kill failed: No such process
ERROR: MDPI driver: Abrupt exception: 6
ERROR: MDPI driver: Unable to send signal.
kill failed: No such process
ERROR: MDPI driver: Abrupt exception: 6
ERROR: MDPI driver: Unable to send signal.
kill failed: No such process
ERROR: MDPI driver: Abrupt exception: 6
ERROR: MDPI driver: Unable to send signal.
kill failed: No such process
ERROR: MDPI driver: Abrupt exception: 6
ERROR: MDPI driver: Unable to send signal.
kill failed: No such process
ERROR: MDPI driver: Abrupt exception: 6
ERROR: MDPI driver: Unable to send signal.
kill failed: No such process
Warning: Returned error code!
error -> The simulation does not end correctly
````

Reply:
I disable parallel building, that should fix the broken result, but now the build hits the 1 hour timeout for ofborg builds.

Reply:
Parallel building with to many tasks can apparently lead to a broken build. This was not an issue for me when building locally, but ofborg has way more cores than I do.

Disabling parallel builds solved that issue. However, ofborg builds now exceed the one hour timelimit which causes them to fail. This is just a ofborg problem, the generated binaries work fine and pass the check.

Reply:
You can do

```
if (( $NIX_BUILD_CORES > X)); then
  NIX_BUILD_CORES=X
fi
```
replace X with a core amount you think will be safe

in `preConfigure` or `preBuild` to keep the core amount at a safe number